### ISSUE TITLE
[java] AddEmptyString: False positive reported for constants

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#6327]
**Affects PMD Version:** 
4.0 - 7.19

**Rule:**

[AddEmptyString](https://pmd.github.io/pmd/pmd_rules_java_performance.html#addemptystring)

**Description:**

The description claims that `Integer.toString(1234)` and `String.valueOf(1234)` are more performant then `1234 + ""`.
From a quick test with recent `javac` that's not always true, since `String.valueOf` will stay in the bytecode and `1234 + ""` may be inlined as "1234".

The rule definitely makes sense when adding an empty string to another string, that's cluttering the code. In most cases the performance gain is questionable and for compile time constants specifically the rule is likely counterproductive. 

**Code Sample demonstrating the issue:**

```java
public class Test {
  public static void duration = 86400;

  void main(Request request) {
     request.setHeader("Cache", duration + "");
  }
}
```

**Expected outcome:**

PMD reports a violation at line 5, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by lukasgraef** [comment]
I did a quick JMH Benchmark ([AddEmptyStringBenchmark.java](https://github.com/user-attachments/files/29255860/AddEmptyStringBenchmark.java)) to check whether this rule's suggestions are even helpful in terms of performance (running on Java 21):

```
Benchmark                                         Mode  Cnt  Score    Error   Units
AddEmptyStringBenchmark.ConstantIntegerConcat    thrpt    5  2.643 ±  0.019  ops/ns
AddEmptyStringBenchmark.ConstantIntegerToString  thrpt    5  0.137 ±  0.020  ops/ns
AddEmptyStringBenchmark.RandomIntegerConcat      thrpt    5  0.037 ±  0.003  ops/ns
AddEmptyStringBenchmark.RandomIntegerToString    thrpt    5  0.038 ±  0.001  ops/ns
```

Apart from verifying the performance loss when using something like `Integer.toString(1234)` compared to `"" + 1234`, the difference in performance between String concat and toString for non-constant values is in the range of error.

To me it looks like there's no reason anymore for `AddEmptyString` being in the "Performance" rule category, unless it's still sensible for older JDKs.
Which category should we move it then? Or maybe we should deprecate this rule entirely?

**Comment by adangel** [comment]
Maybe codestyle? But then we need a reason, why "Integer.toString(1)" should be preferred over """ + 1". It could be readability maybe? But otherwise, I'd be in favor of deprecating the rule.
